pkg/config/config.go

```diff
@@ -150,6 +173,20 @@ func (c *Config) SetConfigFromEnv() error {
 		c.CtxTimeOut = time.Duration(i) * time.Second
 	}
 
+	if gitContainerTemplate := os.Getenv(gitContainerTemplateEnvVar); gitContainerTemplate != "" {
+		c.GitContainerTemplate = corev1.Container{}
+		if err := json.Unmarshal([]byte(gitContainerTemplate), &c.GitContainerTemplate); err != nil {
+			return err
+		}
+		if c.GitContainerTemplate.Image == "" {
+			c.GitContainerTemplate.Image = gitDefaultImage
```

Comment:
> on the same line of thinking here, would you like to enforce the SecurityContext with nonRoot users, in case the definition of a container template does not specify that?

No, actually not. If somebody modifies the template, then I assume this person = administrator has made his/her mind on why and which settings to configure. Potentially that person also applies configuration to Tekton, for example a pod template including a pod-level security context. It would be kind of unforeseeable for him if we then apply an own security context.

Also, the image is mandatory = without it, a container spec is invalid and would for sure not work. It therefore makes sense to ensure one. A security context on the other hand is optional.